tests/input_data/configuration/test_form.py

```diff
@@ -1,29 +1,102 @@
+    f = DummyForm({"prior": {"prior_type": "value", "density": "uniform", "min": 10, "max": 0}})
     assert f.validate_and_normalize() == [
-        ("", "", "Parameters incompatible with density 'uniform': Bounds are inconsistent: lower=10.0 mean=5.0 upper=0.0")
+        (
+            "prior",
+            "prior",
+            "Parameters incompatible with density 'uniform': Bounds are inconsistent: lower=10.0 mean=5.0 upper=0.0",
+        )
     ]
 
-    f = SmoothingPrior({"prior_type": "value", "density": "gaussian", "mean": 0, "std": -1})
+    f = DummyForm({"prior": {"prior_type": "value", "density": "gaussian", "mean": 0, "std": -1}})
     assert f.validate_and_normalize() == [
         (
-            "",
-            "",
+            "prior",
+            "prior",
             "Parameters incompatible with density 'gaussian':"
             " Standard deviation must be positive: standard deviation=-1.0",
         )
     ]
+
+
+def test_SmoothingPrior__global_eta():
+    f = DummyForm(
+        {
+            "eta": {"priors": 0.001, "data": 0.001},
+            "prior": {"prior_type": "value", "density": "uniform", "min": 0, "max": 10},
+        }
+    )
+    f.validate_and_normalize()
+    assert f.prior.prior_object == priors.Uniform(0, 10, eta=0.001)
+
+    f = DummyForm(
+        {
+            "eta": {"priors": 0.001, "data": 0.001},
+            "prior": {"prior_type": "value", "density": "uniform", "min": 0, "max": 10, "eta": 0.002},
+        }
+    )
+    f.validate_and_normalize()
+    assert f.prior.prior_object == priors.Uniform(0, 10, eta=0.002)
+
+
+def test_SmoothingPrior__global_nu():
+    f = DummyForm(
+        {
+            "students_dof": {"priors": 5, "data": 5},
+            "prior": {"prior_type": "value", "density": "students", "mean": 0, "std": 0.1},
+        }
+    )
+    f.validate_and_normalize()
+    assert f.prior.prior_object == priors.StudentsT(0, 0.1, 5)
+
+    f = DummyForm(
+        {
+            "students_dof": {"priors": 5, "data": 5},
+            "prior": {"prior_type": "value", "density": "students", "mean": 0, "std": 0.1, "nu": 3},
+        }
+    )
+    f.validate_and_normalize()
+    assert f.prior.prior_object == priors.StudentsT(0, 0.1, 3)
+
+    f = DummyForm(
+        {
+            "students_dof": {"priors": 5, "data": 5},
+            "prior": {"prior_type": "value", "density": "log_students", "mean": 0, "std": 0.1},
+        }
+    )
+    assert set(f.validate_and_normalize()) == {
+        ("prior", "prior", "Parameters incompatible with density 'log_students': Nu must be positive: nu=None")
+    }
+
+    f = DummyForm(
+        {
+            "log_students_dof": {"priors": 5, "data": 5},
+            "prior": {"prior_type": "value", "density": "students", "mean": 0, "std": 0.1},
+        }
+    )
+    assert set(f.validate_and_normalize()) == {
+        ("prior", "prior", "Parameters incompatible with density 'students': Nu must be positive: nu=None")
```

Comment:
is the density here "log_students_dof" (line 96), or "students" (line 97)?

What I am saying is that in line 101, the incompatibility would be with the "log students" not with "students" as it says? given that in the examples above, it appears that nu does not have to be supplied for "students"?

Reply:
The error is that there is a students-t distribution which did not have it's own nu and there was a global nu for log students-t but not one for students-t.

students-t and log students-t must always have a nu, but it can be supplied by either type specific global settings or local settings, which is what this test exercises.

I'll add some comments because it isn't clear.